mlir/include/mlir/Dialect/Bufferization/IR/InferDestinationOpInterface.td

```diff
@@ -0,0 +1,103 @@
+//===-- InferDestinationOpInterface.td - Infer destination -*- tablegen -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef INFER_DESTINATION_OP_INTERFACE
+#define INFER_DESTINATION_OP_INTERFACE
+
+include "mlir/IR/OpBase.td"
+
+def InferDestinationOpInterface : OpInterface<"InferDestinationOpInterface"> {
+  let description = [{
+    This interface can be implemented by ops that read data from an "input"
+    tensor and store the result into an "output"/"init" tensor (i.e., the
+    "destination" tensor). It drives the empty tensor elimination pass.
```

Comment:
would you say the tensor elimination pass would be the only use for this interface? Or do you think this will have other uses later?

Reply:
The original version of this interface was used only for empty tensor elimination. The new SubsetOpInterface is also used by One-Shot Analysis. In the future, it could also be used by the tiling infrastructure (and then move to `mlir/Interfaces`).